### DISCUSSION TITLE
"Load Error FAILED" - Supported extensions are spec, qontract, wsdl, yaml, yml, json.

### DISCUSSION THREAD

**Discussion by igor-martins-ugowork** [specmatic/specmatic#1039] [Ask For Help]
Hi, I am implementing Specmatic in an existing ReactJS Project. Inside the project I have my stubs and my tests directory, each one with a .yaml file. Whe I run my stub, it works normally, but when I run my test it gives the error below. I am with a .yaml file in my directory of tests, so I have no idea about what is going on. Could anyone help?

> specmatic test src\specmatic\tests

[specmatic] 14-03-2024 13:31:06 info: CLI: Running with args "test src\specmatic\tests"
[specmatic] 14-03-2024 13:31:06 debug: CLI: Specmatic jar path: .\node_modules\specmatic\specmatic.jar
Load Error has FAILED

Reason: Current file src\specmatic\tests has an unsupported extension . Supported extensions are spec, qontract, wsdl, yaml, yml, json.

Unsuccessful Scenarios:

  "Load Error FAILED"
        Reason: Current file src\specmatic\tests has an unsupported extension . Supported extensions are spec, qontract, wsdl, yaml, yml, json.

Tests run: 1, Successes: 0, Failures: 1, Errors: 0

**Comment by joelrosario** [comment]
Hi @igor-martins-ugowork, sorry for the delay, and thanks for getting in touch.

Can you try once again with the latest version of the `specmatic` node module, version `1.3.10`. I believe that it should resolve the error you are seeing.

If the error persists, please let us know.

**Reply by igor-martins-ugowork** [reply]
Hi @joelrosario , thank you for the response. I followed your indication. Firstly, I installed the latest version, which was 1.3.12 today. After that, I installed the 1.3.10, as you wrote. However, for both cases the error persists. When I run my stub, it works, but when I run my test, it breaks. Despite having the .yaml file in my directory of tests, my console displays:

![image](https://github.com/znsio/specmatic/assets/150720524/caa1dd33-5fb6-4093-b0c6-4fc9432fecbe)

I am organizing this way:

![image](https://github.com/znsio/specmatic/assets/150720524/36a24d24-b546-42dd-8982-859302f36f17)

I have this scripts in my package.json

    "spec:stub": "specmatic stub src\\specmatic\\stubs",
    "spec:test": "specmatic test src\\specmatic\\tests",

**Reply by joelrosario** [reply]
Hi @igor-martins-ugowork,

Thanks for sharing additional details, including the contents of your `package.json` file, and your directory structure. It has helped us understand your issue.

Here is how you can setup `scripts` in `package.json`:

```
"spec:stub": "specmatic stub",
"spec:test": "specmatic test",
```

When no specifications are passed as arguments, Specmatic looks for them in it's own configuration, which you will need to now create.

Just create a file in the root folder of the project, named `specmatic.json`. You can read more about it [here](https://specmatic.in/documentation/specmatic_json.html#getting-started).

To start with, `specmatic.json` in your project would probably look like this:

```json
{
  "sources": [
    {
      "provider": "filesystem",
      "test": [
        "src\\specmatic\\tests\\usersF.yaml"
      ],
      "stub": [
        "src\\specmatic\\tests\\users.yaml"
      ]
    }
  ]
}
```

In future, when you want to run tests from other specifications against the same service, or stub out more dependencies, you may add them to the test and stub lists respectively.

Hope this helps.

**Reply by yogeshnikam671** [reply]
Hey @igor-martins-ugowork, are you still facing this issue? Let us know so that we can resolve the same.

**Reply by igor-martins-ugowork** [reply]
Hi @yogeshnikam671 , the solution proposed worked, but I stopped working with Specmatic because it is not the best approach to test my frontend app. So, I didn't explored anymore.
